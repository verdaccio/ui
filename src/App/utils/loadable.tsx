import React, { lazy, Suspense } from 'react';
export default (importCallback: any) => {
  const TargetComponent = lazy(importCallback);
  return (props: any) => (
    <Suspense fallback={null}>
      <TargetComponent {...props} />
    </Suspense>
  );
};
