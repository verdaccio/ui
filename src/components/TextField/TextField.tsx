import { default as MaterialUITextField, TextFieldProps } from '@material-ui/core/TextField';
import React, { forwardRef } from 'react';

// The default element type of MUI's TextField is 'div'
type TextFieldRef = HTMLDivElement;

const TextField = forwardRef<TextFieldRef, TextFieldProps>(function TextField({ InputProps, classes, ...props }, ref) {
  return (
    <MaterialUITextField
      {...props}
      innerRef={ref}
      InputProps={{
        ...InputProps,
        classes,
      }}
    />
  );
});

export default TextField;
