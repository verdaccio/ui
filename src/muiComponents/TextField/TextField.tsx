import React, { forwardRef } from 'react';
import { default as MaterialUITextField, TextFieldProps } from '@material-ui/core/TextField';

// The default element type of MUI's TextField is 'div'
type TextFieldRef = HTMLElementTagNameMap['div'];

const TextField = forwardRef<TextFieldRef, TextFieldProps>(function ToolTip({ InputProps, classes, ...props }, ref) {
  return (
    <MaterialUITextField
      {...props}
      innerRef={ref}
      InputProps={{
        ...InputProps,
        classes,
      }}
    />
  );
});

export default TextField;
