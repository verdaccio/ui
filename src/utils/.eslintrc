{
    "rules": {
        "no-invalid-this": 0
    }
}