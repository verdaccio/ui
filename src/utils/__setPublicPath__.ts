// @ts-ignore
if (!__DEBUG__) {
  // @ts-ignore
  __webpack_public_path__ = window.VERDACCIO_API_URL.replace(/\/verdaccio\/$/, '/static/'); // eslint-disable-line
}
