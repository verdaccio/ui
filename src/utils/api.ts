import storage from './storage';
import '../../types';

/**
 * Handles response according to content type
 * @param {object} response
 * @returns {promise}
 */
export function handleResponseType(response: Response): Promise<[boolean, any]> {
  if (response.headers) {
    const contentType = response.headers.get('Content-Type');
    if (contentType && contentType.includes('application/pdf')) {
      return Promise.all([response.ok, response.blob()]);
    }
    if (contentType && contentType.includes('application/json')) {
      return Promise.all([response.ok, response.json()]);
    }
    // it includes all text types
    if (contentType && contentType.includes('text/')) {
      return Promise.all([response.ok, response.text()]);
    }

    // unfortunately on download files there is no header available
    if (response.url && response.url.endsWith('.tgz') === true) {
      return Promise.all([response.ok, response.blob()]);
    }
  }

  return Promise.all([response.ok, response.text()]);
}

class API {
  public request<T>(url: string, method = 'GET', options: RequestInit = { headers: {} }): Promise<T> {
    if (!window.VERDACCIO_API_URL) {
      throw new Error('VERDACCIO_API_URL is not defined!');
    }

    const token = storage.getItem('token');
    const headers = new Headers(options.headers);

    if (token && headers.has('Authorization') === false) {
      headers.set('Authorization', `Bearer ${token}`);
      options.headers = headers;
    }

    if (!['http://', 'https://', '//'].some(prefix => url.startsWith(prefix))) {
      url = window.VERDACCIO_API_URL + url;
    }

    return new Promise((resolve, reject) => {
      fetch(url, {
        method,
        credentials: 'same-origin',
        ...options,
      })
        .then(handleResponseType)
        .then(response => {
          if (response[0]) {
            resolve(response[1]);
          } else {
            reject(new Error('something went wrong'));
          }
        })
        .catch(error => {
          reject(error);
        });
    });
  }
}

export default new API();
