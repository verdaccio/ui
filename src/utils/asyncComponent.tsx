import React from 'react';

export function asyncComponent(getComponent) {
  return class AsyncComponent extends React.Component {
    static Component = null;
    state = { Component: AsyncComponent.Component };

    componentDidMount() {
      const { Component } = this.state;
      if (!Component) {
        getComponent()
          .then(({ default: Component }) => {
            AsyncComponent.Component = Component;
            /* eslint react/no-did-mount-set-state:0 */
            this.setState({ Component });
          })
          .catch(err => {
            console.error(err);
          });
      }
    }
    render() {
      const { Component } = this.state;
      if (Component) {
        // eslint-disable-next-line verdaccio/jsx-spread
        // @ts-ignore
        return <Component {...this.props} />;
      }

      return null;
    }
  };
}
