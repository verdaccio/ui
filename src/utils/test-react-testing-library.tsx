import { render } from '@testing-library/react';
import React from 'react';
import { I18nextProvider } from 'react-i18next';

import ThemeProvider from 'verdaccio-ui/design-tokens/ThemeProvider';
import APIProvider from 'verdaccio-ui/providers/API/APIProvider';
import AppConfigurationProvider from 'verdaccio-ui/providers/config';

import i18nConfig from '../../i18n/config';

const customRender = (node: React.ReactElement, ...options: any) => {
  return render(
    <AppConfigurationProvider>
      <APIProvider>
        <ThemeProvider>
          <I18nextProvider i18n={i18nConfig}>{node}</I18nextProvider>
        </ThemeProvider>
      </APIProvider>
    </AppConfigurationProvider>,
    ...options
  );
};

export * from '@testing-library/react';
export { customRender as render };
