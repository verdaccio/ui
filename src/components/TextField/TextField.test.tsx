import React from 'react';

import { render } from 'verdaccio-ui/utils/test-react-testing-library';

import TextField from './TextField';

describe('<TextField /> component', () => {
  const props = {
    name: 'test',
    value: 'test',
  };
  test('should load the component in default state', () => {
    const { container } = render(<TextField name={props.name} value={props.value} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
