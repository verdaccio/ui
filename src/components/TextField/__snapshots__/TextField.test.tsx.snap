// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> component should load the component in default state 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root"
>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input"
      name="test"
      type="text"
      value="test"
    />
  </div>
</div>
`;
